### ISSUE TITLE
Module cycle with ARCH_IS_UNKNOWN_ENDIAN

### ISSUE THREAD

**Issue by achirkin** [haskell-foundation/foundation#227]
Trying to build `foundation-0.0.4` from cabal on `arm64` arch I got into module imports cycle:
```
Module imports form a cycle:
         module `Foundation.Array.Unboxed.Mutable`
        imports `Foundation.Primitive.Types`
  which imports `Foundation.Primitive.Endianness`
  which imports `Foundation.System.Info`
  which imports `Foundation.String`
  which imports `Foundation.String.UTF8`
  which imports `Foundation.Array.Unboxed.Mutable`
```

The clue, I guess, is here [`Foundation.Primitive.Endianness`](https://github.com/haskell-foundation/foundation/blob/master/Foundation/Primitive/Endianness.hs#L26). After I changed slightly cabal file to have `ARCH_IS_LITTLE_ENDIAN`, the package compiles just fine.

**Comment by achirkin** [comment]
Thanks!

**Comment by vincenthz** [comment]
thanks for reporting, should be fixed in 0.0.5. we don't have a handy arm64 testing / CI machine for now. hopefully some day.
